src/warpedview.jl

```diff
@@ -1,62 +1,86 @@
-immutable WarpedView{T,N,A<:AbstractArray,F1<:Transformation,F2<:Transformation,I} <: AbstractArray{T,N}
+_build_extrapolation(etp::AbstractExtrapolation) = etp
+
+function _build_extrapolation{T}(itp::AbstractInterpolation{T}, fill::FillType = _default_fill(T))
+    etp = extrapolate(itp, fill)
+    _build_extrapolation(etp)
+end
+
+function _build_extrapolation{T,N,D<:Union{Linear,Constant}}(parent::AbstractArray{T,N}, degree::D = Linear(), args...)
+    itp = Interpolations.BSplineInterpolation{T,N,typeof(parent),BSpline{D},OnGrid,0}(parent)
+    _build_extrapolation(itp, args...)
+end
+
+immutable WarpedView{T,N,A<:AbstractArray,F1<:Transformation,I,F2<:Transformation,E<:AbstractExtrapolation} <: AbstractArray{T,N}
     parent::A
     transform::F1
-    transform_inv::F2
     indices::I
+    inverse::F2
```

Comment:
I did notice the `inv` part quite early. Well, that sounds more observant as it is, because basically I had to reverse engineer what the code does in order to understand what is going on. I also spend some thoughts on the issue you describe.

I like this approach for the simple reason as it behaves how I want it to behave. I like to reason about how the image changes, not how the indices change. Also I do want the option to nest `WarpedView` for a use case I have. Basically I like this way because it is convenient for me.

Why don't we avoid commitment to one interpretation altogether. We could add a `InvWarpedView` (and `invwarp`)  which does `w = img[tfm(x)]` as an alternative.